src/prefs/WaveformSettings.h

```diff
@@ -81,4 +89,34 @@ class AUDACITY_DLL_API WaveformSettings : public PrefsListener
    // Convenience
    bool isLinear() const { return stLinear == scaleType; }
 };
+
+class AUDACITY_DLL_API WaveformSettingsCache
```

Comment:
Something feels really wrong with this class. Judging by comment - it allows changing application global state. Following the real post effects is really hard here too.

Reply:
Which comment do you mea?

This is extra state, but not of the application.  Each WaveTrack has its own independently changeable cache.

Reply:
Specifically this comment is really confusing.  `//! Get either the global default settings, or the track's own if previously created`

Reply:
That comment may be extraneous, yes.  By mistaken analogy with the Spectogram settings code where that comment makes more sense.